Dockerfile

```
@@ -1,112 +1,39 @@
-# 2) pin conda to the version given by CIVIS_CONDA_VERSION
-# 3) install the python version CIVIS_PYTHON_VERSION
-# 4) pin the python version
-#
-# Extra symlinks are added at the end because...
-#   Red Hat and Debian use different names for this file. git2R wants the latter.
-#   See conda-recipes GH 423
-RUN echo 'export PATH=/opt/conda/bin:$PATH' > /etc/profile.d/conda.sh && \
-    wget --quiet https://repo.continuum.io/miniconda/Miniconda3-${CIVIS_MINICONDA_VERSION}-Linux-x86_64.sh && \
-    /bin/bash /Miniconda3-${CIVIS_MINICONDA_VERSION}-Linux-x86_64.sh -b -p /opt/conda && \
-    rm Miniconda3-${CIVIS_MINICONDA_VERSION}-Linux-x86_64.sh && \
-    /opt/conda/bin/conda install --yes conda==${CIVIS_MINICONDA_VERSION} && \
-    conda install conda=${CIVIS_CONDA_VERSION} && \
-    echo "conda ==${CIVIS_CONDA_VERSION}" > /opt/conda/conda-meta/pinned && \
-    conda install --yes python==${CIVIS_PYTHON_VERSION} && \
-    echo "python ==${CIVIS_PYTHON_VERSION}" >> /opt/conda/conda-meta/pinned && \
-    conda clean --all -y && \
-    ln -s /opt/conda/lib/libopenblas.so /opt/conda/lib/libblas.so && \
-    ln -s /opt/conda/lib/libopenblas.so /opt/conda/lib/liblapack.so && \
-    ln -s /opt/conda/lib/libssl.so /opt/conda/lib/libssl.so.6 && \
-    ln -s /opt/conda/lib/libcrypto.so /opt/conda/lib/libcrypto.so.6
-
-# Install boto in the base environment for private s3 channel support.
-# Install Python Packages
-COPY .condarc /opt/conda/.condarc
-COPY environment.yml environment.yml
-RUN conda install -y boto && \
-    conda install -y nomkl && \
-    conda env update -f environment.yml -n root && \
-    conda clean --all -y && \
-    rm -rf ~/.cache/pip
-
-# We aren't running a GUI, so force matplotlib to use
-# the non-interactive "Agg" backend for graphics.
-# Run matplotlib once to build the font cache.
-ENV MATPLOTLIBRC=${HOME}/.config/matplotlib/matplotlibrc
-RUN mkdir -p ${HOME}/.config/matplotlib && \
-    echo "backend      : Agg" > ${HOME}/.config/matplotlib/matplotlibrc && \
-    python -c "import matplotlib.pyplot"
+COPY requirements-full.txt .
 
-# Enable widgetsnbextension for jupyter widgets.
-# See https://ipywidgets.readthedocs.io/en/stable/user_install.html.
-# This enables the extension in the conda environment. The conda-forge version
-# does this upon installation, but the default channel version doesn't seem to,
-# so we'll run this (again) just in case.
-RUN jupyter nbextension enable --py widgetsnbextension
+RUN pip install -r requirements-full.txt
 
 # Instruct joblib to use disk for temporary files. Joblib defaults to
 # /shm when that directory is present. In the Docker container, /shm is
 # present but defaults to 64 MB.
 # https://github.com/joblib/joblib/blob/0.11/joblib/parallel.py#L328L342
 ENV JOBLIB_TEMP_FOLDER=/tmp
 
-ENV VERSION=6.5.0 \
-    VERSION_MAJOR=6 \
-    VERSION_MINOR=5 \
-    VERSION_MICRO=0
+ENV VERSION=6.6.0 \
+  VERSION_MAJOR=6 \
+  VERSION_MINOR=6 \
+  VERSION_MICRO=0
```

This rewrite is definitely a breaking change! A new entry in [`CHANGELOG.md`](https://github.com/civisanalytics/datascience-python/blob/master/CHANGELOG.md) is also needed (where I think we can simply mention the Python 3.11 upgrade, no more conda, and just keeping the core data science packages, probably no need to list the exact version number changes for a complete rewrite like this).

```suggestion
ENV VERSION=7.0.0 \
  VERSION_MAJOR=7 \
  VERSION_MINOR=0 \
  VERSION_MICRO=0
```